lockfile_unix.go

```diff
@@ -37,24 +39,40 @@ type lockfile struct {
 	lw       string
 	locktype int16
 	locked   bool
+	ro       bool
 }
 
-// Lock locks the lock file
-func (l *lockfile) Lock() {
+func (l *lockfile) lock(l_type int16, l_cmd int) {
 	lk := unix.Flock_t{
-		Type:   l.locktype,
+		Type:   l_type,
 		Whence: int16(os.SEEK_SET),
 		Start:  0,
 		Len:    0,
 		Pid:    int32(os.Getpid()),
 	}
-	l.mu.Lock()
-	l.locked = true
+	if !l.ro {
```

Comment:
So we only lock with write lock?

Reply:
Also testing but, if we have a read-only lock, then I don't see why we need to have mutex which would serialize reads of the same process.